File: src/app/src/main/java/com/onscreensync/tvapp/utils/JsonUtils.java
package com.onscreensync.tvapp.utils;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

public class JsonUtils {
    private static final Gson gson = new Gson();

    // Generic method for deserialization
    public static <T> T fromJson(String jsonString, Class<T> classOfT) {
        if(ObjectExtensions.isNullOrEmpty(jsonString)) return null;

        try {
            return gson.fromJson(jsonString, classOfT);
        } catch (JsonSyntaxException e) {
            return null;
        }
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/utils/JsonUtils.kt
package com.onscreensync.tvapp.utils

import com.google.gson.Gson
import com.google.gson.JsonSyntaxException

object JsonUtils {
    private val gson = Gson()

    @JvmStatic
    fun <T> fromJson(jsonString: String?, classOfT: Class<T>): T? {
        if (jsonString.isNullOrEmpty()) return null

        return try {
            gson.fromJson(jsonString, classOfT)
        } catch (e: JsonSyntaxException) {
            null
        }
    }
}